plugin.py

```diff
@@ -783,16 +779,18 @@ def doPrivmsg(self, irc, msg):
                 irc.reply(admonition)
 
     def get_current_release(self):
-        url = 'https://admin.fedoraproject.org/pkgdb/api/collections/'
-        query = {
-            'clt_status': 'Active',
-            'pattern': 'f*',
-        }
-        response = requests.get(url, params=query)
+        url = 'https://pdc.fedoraproject.org/rest_api/v1/releases/'\
+              '?active=true&name=Fedora&release_type=ga&fields=version'\
+              '&ordering=version'
+        response = requests.get(url)
         data = response.json()
-        collections = data['collections']
-        collections.sort(key=lambda c: int(c['version']))
-        return collections[-1]['branchname'].encode('utf-8')
+        return str(
+            max(
+                map(
+                    lambda x: int(x['version']),
+                    filter(
+                        lambda x: x['version'] != 'Rawhide',
+                        data['results']))))
```

Comment:
Can we change this to something like this?

```
        str(max(map(lambda x: int(x['version']),
                    filter(lambda x: x['version'] != 'Rawhide',
                           data['results'])))
```

Reply:
tbh, I'd encourage dropping the lambda to something else

Reply:
@pypingou such as?

Reply:
@sayanchowdhury changed

Reply:
@relrod maybe just a regular function?

Reply:
@pypingou but the lambdas would translate into one line functions...

```python
def is_version_rawhide(x):
    return x['version'] != 'Rawhide'

def int_version(x):
    return int(x['version'])
```

I'm happy to change it if you want me to, but to me that just seems more verbose and less clear.

Reply:
To be clear, the difference is:

```python
        return str(max(map(lambda x: int(x['version']),
                           filter(lambda x: x['version'] != 'Rawhide',
                                  data['results']))))
```

versus

```python
def is_version_rawhide(x):
    return x['version'] != 'Rawhide'

def int_version(x):
    return int(x['version'])

# ...

    return str(max(map(int_version,
                       filter(is_version_rawhide,
                              data['results']))))
```

Reply:
@pypingou I changed it to avoid the issue altogether and use a list comprehension instead...

Reply:
@relrod I was thinking it may work (I meant to double checked before suggesting it), thanks for doing it, looks much clearer to me now :)